src/parser/core/modules/Ability.ts

```diff
@@ -29,9 +29,12 @@ export interface SpellbookAbility<TrackedAbilityType extends TrackedAbility = Tr
    */
   category: string;
   /**
-   * The cooldown of a spell at the time of the cast, this can be a function
-   * for more complicated calls or even to check for buffs. Parameters
-   * provided: `hastePercentage`, `selectedCombatant`
+   * The cooldown of a spell at the time of the cast. Unlike most other durations in WoWA,
+   * this is in *seconds, NOT milliseconds*. This can be the direct number, or it can be a function
+   * for cooldowns that scale with haste, and can also branch based on combatantinfo.
+   * @param haste the player's haste at the time of spell cast, as a proportion of
+   *   normal casting speed (20% haste will be passed as 1.20).
+   * @param trigger TODO what the hell is this and what is it for?
    */
   cooldown?: ((haste: number, trigger?: AnyEvent) => number) | number;
```

Comment:
I still can't figure out who wants the `trigger` in cooldown

Reply:
no tests fail from removing it, and typecheck passes with minimal changes (removing references to the parameter from the `Ability` and `Abilities` modules in core). I'm good removing it.